charts/scalardb-analytics-postgresql/templates/scalardb-analytics-postgresql/deployment.yaml

```diff
@@ -0,0 +1,154 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "scalardb-analytics-postgresql.fullname" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "scalardb-analytics-postgresql.labels" . | nindent 4 }}
+spec:
+  minReadySeconds: 0
+  replicas: {{ .Values.scalardbAnalyticsPostgreSQL.replicaCount }}
+  selector:
+    matchLabels:
+      {{- include "scalardb-analytics-postgresql.selectorLabels" . | nindent 6 }}
+  {{- with .Values.scalardbAnalyticsPostgreSQL.strategy }}
+  strategy:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+  template:
+    metadata:
+      annotations:
+        checksum/config: {{ include (print $.Template.BasePath "/scalardb-analytics-postgresql/configmap.yaml") . | sha256sum }}
+        {{- if .Values.scalardbAnalyticsPostgreSQL.podAnnotations }}
+        {{- toYaml .Values.scalardbAnalyticsPostgreSQL.podAnnotations | nindent 8 }}
+        {{- end }}
+      labels:
+        {{- include "scalardb-analytics-postgresql.selectorLabels" . | nindent 8 }}
+    spec:
+      restartPolicy: Always
+      {{- if .Values.scalardbAnalyticsPostgreSQL.serviceAccount.serviceAccountName }}
+      serviceAccountName: {{ .Values.scalardbAnalyticsPostgreSQL.serviceAccount.serviceAccountName }}
+      {{- end }}
+      automountServiceAccountToken: {{ .Values.scalardbAnalyticsPostgreSQL.serviceAccount.automountServiceAccountToken }}
+      terminationGracePeriodSeconds: 60
+      {{- with .Values.scalardbAnalyticsPostgreSQL.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      securityContext:
+        {{- toYaml .Values.scalardbAnalyticsPostgreSQL.podSecurityContext | nindent 8 }}
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.scalardbAnalyticsPostgreSQL.securityContext | nindent 12 }}
+          image: "{{ .Values.scalardbAnalyticsPostgreSQL.image.repository }}:{{ .Values.scalardbAnalyticsPostgreSQL.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.scalardbAnalyticsPostgreSQL.image.pullPolicy }}
+          ports:
+            - containerPort: 5432
+          env:
+          - name: POSTGRES_DB
+            value: "{{ .Values.scalardbAnalyticsPostgreSQL.postgresql.databaseName }}"
+          envFrom:
+          - secretRef:
+              name:  "{{ .Values.scalardbAnalyticsPostgreSQL.postgresql.secretName }}"
```

Comment:
We set the superuser's password of PostgreSQL via a Secret resource.

Reply:
I prefer to use `env` because I don't know the key name that should be included in the secret from envFrom (although it is case by case).

Reply:
Ah, I see. In this case, the environment variable name is fixed `POSTGRES_PASSWORD`. It depends on the [PostgreSQL official container image](https://hub.docker.com/_/postgres).

In other words, users cannot use arbitrary environment variable names, and there is no special reason that I use `envFrom` here.

So, I will update to use `env`. Thank you for your suggestion!

Reply:
Fixed in 8af1a27313f53b9c07d1e36a76d6127f210c38bc.